src/main/java/frc/robot/subsystems/spinner/Spinner.java

```diff
@@ -0,0 +1,133 @@
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import com.revrobotics.ColorMatch;
+import com.revrobotics.ColorMatchResult;
+import com.revrobotics.ColorSensorV3;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.util.Color;
+import frc.robot.constants.RobotMap;
+import frc.robot.subsystems.OverridableSubsystem;
+import frc.robot.utils.DriverStationLogger;
+import frc.robot.constants.RobotConstants.SpinnerConstants;
+
+public class Spinner extends OverridableSubsystem {
+  private final WPI_TalonSRX talonSRX;
+  private final ColorSensorV3 colorSensor;
+  private final ColorMatch colorMatcher;
+
+  /**
+   * This class holds all of the methods for the trench roller subsystem, which
+   * spins the trench.
+   */
+  public Spinner() {
+    talonSRX = new WPI_TalonSRX(RobotMap.kRollerTalonSRX);
+    talonSRX.setNeutralMode(NeutralMode.Coast);
+    talonSRX.setInverted(SpinnerConstants.kIsInverted);
+    talonSRX.configOpenloopRamp(SpinnerConstants.kRampRate);
+    talonSRX.configClosedloopRamp(SpinnerConstants.kRampRate);
+
+    colorSensor = new ColorSensorV3(RobotMap.kI2cPort);
+    colorMatcher = new ColorMatch();
+
+    createColors();
+  }
+
+  @Override
+  public void overriddenMove(double power) {
+    talonSRX.set(power);
+  }
+
+  /**
+   * @return an enum of the color, including unknown if the minimum threshold is
+   *         not met.
+   */
+  public Color getColor() {
+    return colorSensor.getColor();
+  }
+
+  /**
+   * Compare two colors
+   * 
+   * @return are the two colors equal.
+   */
+  public boolean compareColors(Color color, Color comparedColor) {
+    ColorMatchResult matchResult = colorMatcher.matchClosestColor(color);
+    return matchResult.color == comparedColor ? true : false;
+  }
+
+  public boolean isOnColor(Color desiredColor) {
+    return compareColors(colorSensor.getColor(), desiredColor);
+  }
+
+  /**
+   * @return Proximity measurement value, ranging from 0 - far, to 2047 - close.
+   */
+  public int getProximity() {
+    return colorSensor.getProximity();
+  }
+
+  public Color getFMSColor() {
+    String gameData;
+    gameData = DriverStation.getInstance().getGameSpecificMessage();
+    if (gameData.length() > 0) {
+      switch (gameData.charAt(0)) {
+      case 'B':
+        return Color.kFirstBlue;
+      case 'G':
+        return Color.kGreen;
+      case 'R':
+        return Color.kFirstRed;
+      case 'Y':
+        return Color.kYellow;
+      default:
+        DriverStationLogger.logErrorToDS("FMS COLOR DATA IS CORRUPT");
+        return null;
+      }
+    } else {
+      DriverStationLogger.logToDS("Did not get any fms data");
+      return null;
+    }
+  }
+
+  /** @return True when the desired color is within two of the current color */
+  public int calculateSpinDirection(Color currentColor, Color desiredColor) {
+    int currentColorIndex = 0;
+    int desiredColorIndex = 0;
+    for (int i = 0; i >= Colors.values().length - 1; i++) {
```

Comment:
loop ends immediately, change it to:
``for (int i = 0; i < Colors.values().length; i++) ``